splunk-otel-android/src/main/java/com/splunk/rum/RuntimeDetails.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.rum;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.os.BatteryManager;
+import androidx.annotation.Nullable;
+import java.io.File;
+import java.util.concurrent.atomic.AtomicReference;
+
+/** Represents details about the runtime environment at a time */
+final class RuntimeDetails extends BroadcastReceiver {
+
+    private final AtomicReference<Double> batteryPercent = new AtomicReference<>();
```

Comment:
I thought that with certain access patterns that `Atomic*` can sometimes outperform `volatile`  because there are separate read and write locks? I'm not sure that's relevant here, and the volatile is probably simpler so I'll change it.